src/monitord/generate_reports.c

```diff
@@ -46,33 +46,32 @@ void generate_reports(int cday, int cmon, int cyear, const struct tm *p)
                 s++;
                 continue;
             } else if (pid == 0) {
+                /* Child process */
                 char fname[256];
                 char aname[256];
                 fname[255] = '\0';
                 aname[255] = '\0';
                 snprintf(fname, 255, "/logs/.report-%d.log", (int)getpid());
 
                 minfo("Starting daily reporting for '%s'", mond.reports[s]->title);
-                mond.reports[s]->r_filter.fp = wfopen(fname, "w+");
-                if (!mond.reports[s]->r_filter.fp) {
-                    merror("Unable to open temporary reports file.");
-                    s++;
-                    continue;
-                }
+                mdebug1("Generating report from PID %d", (int)getpid());
 
                 snprintf(aname, 255, "%s/%d/%s/ossec-%s-%02d.log", ALERTS, cyear, monthss[cmon], "alerts", cday);
 
                 for (i = 1; !IsFile(aname); i++) {
-                    /* Open the log file */
-                    snprintf(aname, 255, "%s/%d/%s/ossec-%s-%02d.log", ALERTS, cyear, monthss[cmon], "alerts", cday);
+                    mond.reports[s]->r_filter.fp = wfopen(fname, "w+");
+                    if (!mond.reports[s]->r_filter.fp) {
+                        merror("Unable to open temporary reports file.");
+                        exit(1);
+                    }
```

Comment:
Let me know if I'm get the changes right. 

The fopen call was moved here due to the close in the line 89, and it covers the case we have multiple rotated logs in the same day, that I think also was not working due to that duplicate line overwriting aname. 

 The replacement of `continue` by `exit(1)` was to prevent the child to continue and create another child? I think this helps the functionality if we have other elements in the array reports[s] because it will terminate the child that failed and create another child for the new report.

Reply:
> The fopen call was moved here due to the close in the line 89, and it covers the case we have multiple rotated logs in the same day, that I think also was not working due to that duplicate line overwriting aname.

You are right. It was moved because the feature wasn't working anyway. 

> The replacement of continue by exit(1) was to prevent the child to continue and create another child? 

Yes, the child must end on error, not keep iterating the configuration structure.